src/js/components/Header/Header.js

```diff
@@ -1,19 +1,86 @@
-import React from 'react';
-
+import React, { useEffect, useState } from 'react';
 import { Box } from '../Box';
+import { useForwardedRef } from '../../utils';
+
+const Header = React.forwardRef(({ sticky, ...rest }, ref) => {
+  const containerRef = useForwardedRef(ref);
+  const [stickyStyles, setStickyStyles] = useState();
+  const [scrollUp, setScrollUp] = useState(false);
+
+  useEffect(() => {
+    // threshold inital value set to 0
+    const threshold = 0;
+    // number of pixels the document is currently scrolled
+    let lastScrollY = window.pageYOffset;
+    // run event listener callback once
+    let runEventListener = false;
+
+    const updateScrollDir = () => {
+      const scrollY = window.pageYOffset;
+
+      if (Math.abs(scrollY - lastScrollY) < threshold) {
+        runEventListener = false;
+        return;
+      }
+      if (scrollY === 0) {
+        setScrollUp(false);
+        runEventListener = false;
+      } else setScrollUp(scrollY < lastScrollY);
+      setStickyStyles({
+        width: `${containerRef.current.getBoundingClientRect().width}px`,
+        height: `${containerRef.current.getBoundingClientRect().height}px`,
+        left: `${containerRef.current.getBoundingClientRect().left}px`,
+        right: `${containerRef.current.getBoundingClientRect().right}px`,
+      });
+      lastScrollY = scrollY > 0 ? scrollY : 0;
+      runEventListener = false;
+    };
+
+    const onScroll = () => {
+      if (!runEventListener) {
+        // calculate the new offset after the page gets rendered
+        window.requestAnimationFrame(updateScrollDir);
+        runEventListener = true;
+      }
+    };
+
+    window.addEventListener('scroll', onScroll);
+
+    return () => window.removeEventListener('scroll', onScroll);
+  }, [containerRef, scrollUp]);
+
+  const stickyStyle = {
+    position: 'fixed',
+    zIndex: '1',
```

Comment:
I wonder if "1" is always going to be correct. For instance, if a Drop is open in the main content that is scrolling under the header, is there a chance the Drop will render above the Header? We should test this.

Reply:
Good call, Grommet's Layer and Drop have zIndex set to 20 by default

Reply:
good thinking Ill test this out with the Drop then if that is the case I can use the 20 default

Reply:
I think we want to make sure that it is above the Drop, not just equal to it. And, we might want to put this in the theme, especially if the Drop is.

Reply:
sounds good Ill handle this portion of it in the theme I believe Drop is in the Global.Drop do you have a preference where this would live?
`global.header.zIndex`
`theme.header.sticky.zIndex`

Reply:
I'm thinking `theme.header.sticky.zIndex`